### PR TITLE
Updated to use OSS Index instead of a github file

### PR DESCRIPTION
**Pull request by digitalcoyote**
I made some unnecessary changes in addition to changing the vulnerable package source to OSS Index.

- Added a .Net Standard 2.0 version of the task (to support other .Net Runtimes)
   - 2.0 is the lowest version I could target, but should support any higher versions in the future
   - This will need additional work if we want to check .Net Core packages as they are stored in the project file no a packages.config
- Added a .Net Core test project for the .Net Standard version.
  - Unfortunately resources are treated a little differently in .Net Core projects so this is not currently testing the full 

And I made the changes I intended to when I started
- Pulled in the lib from https://github.com/OSSIndex/audit.net/tree/master/src/NugetAuditor
  - Included Audit.net license.txt from their solution for their code.
  - If we can get a NuGet of this, we could just pull it in eventually (the Audit.Net NuGet package is not the same)
- Swapped out unsafepackages.xml logic with a call from NugetAuditor that checks OSS Index

I feel that it may be beneficial in the future to add some additional logic
 - Optional local storage of known unsafe packages
   - No reason to keep making calls out for a package that is known to be vulnerable
   - Option to ignore specific vulnerabilities
   - Ability to add vulnerabilities not reported from OSS Index
- Other applications on GitHub reference a throttling issue that can be resolved by optionally passing credentials

### CODE REVIEW

**Comment by digitalcoyote** [comment]
Addresses #18

File: NugetAuditor/AuditResult.cs [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+using System.Threading.Tasks;
+
+namespace NugetAuditor.Lib
+{
+    public class AuditResult
+    {
+        private PackageId _packageId;
+        private Package _package;
+        
+        public PackageId PackageId
```
Comment (jasond-s) [inline NugetAuditor/AuditResult.cs]: Seems like there are a lot of formatting errors here. Make sure your editor is set to either tabs OR spaces, looks like there's a mix of both.

Comment (digitalcoyote) [reply NugetAuditor/AuditResult.cs]: Running Resharper across the whole solution made a slew of changes it appears the inconsistencies in line endings mostly came from the lib I pulled from Audit.Net, but there were several indentation inconsistencies (various spacing counts on indentation) throughout SafeNuGet as well so it hit a lot of files.

Comment (jasond-s) [reply NugetAuditor/AuditResult.cs]: It is quite a massive set of changes, so maybe a good time to make code style consistent. Not my call, I'm a humble/interested observer, of course.

**Comment by digitalcoyote** [comment]
@paulclarkin I think development in this may have stalled, if you need this change, I can fork and publish to NuGet.

**Comment by digitalcoyote** [comment]
I'm going to rework this to remove unnecessary code and make a few other changes I'd wanted when I first wrote this. I'll be publishing a forked version of this to NuGet when I get a moment (in the process of moving)

**Comment by digitalcoyote** [comment]
On further research into what is required to make this work as an msbuild task, I think I'm going to have to take a different direction to implement this kind of functionality at build. This as it is will not work well.

**Comment by digitalcoyote** [comment]
@paulclarkin  I had to use a bundled .Net tool instead of an msbuild task to get this to work so I just rewrote the whole project. I'm publishing it to [NuGet](https://www.nuget.org/packages/NuGetDefense/). 
If anyone wants to make it work using a pure MSBuild Task like SafeNuGet, You'll need to pack any dependencies in with it and you may want to read [here](https://natemcmaster.com/blog/2017/11/11/msbuild-task-with-dependencies/) before attempting it.

**Pull request closed without merging**